WMF Framework/LocationManager.swift

```diff
@@ -17,30 +17,180 @@ extension LocationManagerConfiguration {
 final public class LocationManager: NSObject, LocationManagerProtocol {
 
     /// Last know location
-    public var location: CLLocation? { fatalError() }
+    public private(set) lazy var location: CLLocation? = self.locationManager.location
+
     /// Last know heading
-    public var heading: CLHeading? { fatalError() }
+    public private(set) lazy var heading: CLHeading? = self.locationManager.heading
+
     /// Return `true` in case when monitoring location, in other case return `false`
     public private(set) var isUpdating = false
+
     /// Delegate for update location manager
     public var delegate: LocationManagerDelegate?
+
     /// Get current locationManager permission state
-    public var autorizationStatus: CLAuthorizationStatus { fatalError() }
-    // TODO: Temporary helper for support ObjC codebase, leter can call directly `autorizationStatus.authorized`
-    /// Return `true` if user is aurthorized or authorized always
-    public var isAuthorized: Bool { fatalError() }
+    public var autorizationStatus: CLAuthorizationStatus { type(of: locationManager).authorizationStatus() }
+
+    // TODO: Temporary helper for support ObjC codebase, leter can call directly `autorizationStatus.authorized`.
+    /// Return `true` if user is aurthorized or authorized always.
+    public var isAuthorized: Bool { autorizationStatus.isAuthorized }
 
     /// Start monitoring location and heading updates.
-    public func startMonitoringLocation() { }
+    public func startMonitoringLocation() {
+        guard autorizationStatus != .notDetermined else {
+            authorize(succcess: startMonitoringLocation)
+            return
+        }
+
+        guard isAuthorized else {
+            authorizedCompletion = startMonitoringLocation
```

Comment:
// Start monitoring location in case the authorization status ever changes to `authorized`